packages/transaction-builder/src/models/msgs/msg-proto-node-stake.ts

```diff
@@ -0,0 +1,105 @@
+import { MsgProtoNodeStake } from './../proto/generated/tx-signer'
+import { Any } from '../proto/generated/google/protobuf/any'
+import { TxMsg } from './tx-msg'
+
+/**
+ * Model representing a MsgNodeStake to stake as an Node in the Pocket Network
+ */
+export class MsgProtoNodeStakeTx extends TxMsg {
+  public readonly KEY: string = '/x.nodes.MsgProtoStake'
+  public readonly AMINO_KEY: string = 'pos/MsgStake'
+  public readonly DEFAULT_PORT: string = '443'
+  public readonly DEFAULT_PROTOCOL: string = 'https:'
+  public readonly pubKey: Buffer
+  public readonly chains: string[]
+  public readonly amount: string
+  public readonly serviceURL: URL
+
+  /**
+   * @param {string} pubKey - Public key
+   * @param {string[]} chains - String array containing a list of blockchain hashes
+   * @param {string} amount - Amount to be sent, has to be a valid number and cannot be lesser than 0
+   * @param {URL} serviceURL - Service node URL, needs to be https://
+   */
+  constructor(
+    pubKey: string,
+    chains: string[],
+    amount: string,
+    serviceURL: URL
+  ) {
+    super()
+    this.pubKey = toUint8Array(pubKey)
+    this.chains = chains
+    this.amount = amount
+    this.serviceURL = serviceURL
+
+    if (!this.serviceURL.port) {
+      this.serviceURL.port = '443'
+    }
+
+    const amountNumber = Number(this.amount) || -1
+    if (isNaN(amountNumber)) {
+      throw new Error('Amount is not a valid number')
+    } else if (amountNumber < 0) {
+      throw new Error('Amount < 0')
+    } else if (this.chains.length === 0) {
+      throw new Error('Chains is empty')
+    }
+  }
+
+  /**
+   * Returns the parsed serviceURL
+   * @returns {string} - Parsed serviceURL
+   * @memberof MsgNodeStake
+   */
+  private getParsedServiceURL(): string {
+    return `${
+      this.serviceURL.protocol
+        ? this.serviceURL.protocol
+        : this.DEFAULT_PROTOCOL
+    }//${this.serviceURL.hostname}:${
+      this.serviceURL.port ? this.serviceURL.port : this.DEFAULT_PORT
+    }`
+  }
+
+  /**
+   * Converts an Msg Object to StdSignDoc
+   * @returns {object} - Msg type key value.
+   * @memberof MsgNodeStake
+   */
+  public toStdSignDocMsgObj(): object {
+    return {
+      type: this.AMINO_KEY,
+      value: {
+        chains: this.chains,
+        public_key: {
+          type: 'crypto/ed25519_public_key',
+          value: this.pubKey.toString('hex'),
+        },
+        service_url: this.getParsedServiceURL(),
+        value: this.amount,
+      },
+    }
+  }
+
+  /**
+   * Converts an Msg Object to StdSignDoc
+   * @returns {any} - Msg type key value.
+   * @memberof MsgNodeStake
+   */
+  public toStdTxMsgObj(): any {
+    const data = {
+      Publickey: this.pubKey,
+      Chains: this.chains,
+      value: this.amount,
+      ServiceUrl: this.getParsedServiceURL(),
+    }
+
+    return Any.fromJSON({
+      typeUrl: this.KEY,
+      value: MsgProtoNodeStake.encode(data).finish().toString('base64'),
```

Comment:
`Uint8Array` and `Buffer` types are basically the same, so it should work either way. Both ways are used in the codebase (see msg-proto-node-unstake), so let's just keep one of them (either using Buffer.from, or not).

```suggestion
      value: Buffer.from(MsgProtoNodeStake.encode(data).finish()).toString('base64'),
```

Reply:
Ah yeah I forgot to add this in the description. For sure we'll only keep one, and I think `Buffer` will stay, but I need to run tests to see how isomorphic `Buffer` actually is compared to the library we're using for `Uint8Array`s. Will fix.

Reply:
Will fix in a new PR after I run tests tomorrow.